signature/mechanism.go

```diff
@@ -2,120 +2,31 @@
 
 package signature
 
-import (
-	"bytes"
-	"fmt"
-
-	"github.com/mtrmac/gpgme"
-)
-
 // SigningMechanism abstracts a way to sign binary blobs and verify their signatures.
+// Each mechanism should eventually be closed by calling Close().
 // FIXME: Eventually expand on keyIdentity (namespace them between mechanisms to
 // eliminate ambiguities, support CA signatures and perhaps other key properties)
 type SigningMechanism interface {
-	// ImportKeysFromBytes imports public keys from the supplied blob and returns their identities.
-	// The blob is assumed to have an appropriate format (the caller is expected to know which one).
-	// NOTE: This may modify long-term state (e.g. key storage in a directory underlying the mechanism).
-	ImportKeysFromBytes(blob []byte) ([]string, error)
-	// Sign creates a (non-detached) signature of input using keyidentity
+	Close() error
+	// SupportsSigning returns nil if the mechanism supports signing, or an error message.
+	SupportsSigning() error
```

Comment:
Added a `SigningNotSupportedError`; in `openpgpSigningMechanism` both `SupportsSigning` and `Sign` now return that type.